python/src/robyn/data/validation/mmmdata_validation.py

```diff
@@ -15,46 +17,176 @@ def check_missing_and_infinite(self) -> ValidationResult:
+        missing_cols = dt_input.columns[dt_input.isnull().any()].tolist()
+        infinite_cols = dt_input.columns[np.isinf(dt_input).any()].tolist()
+        
+        error_details = {}
+        error_message = ""
+        
+        if missing_cols:
+            error_details['missing'] = missing_cols
+            error_message += f"Dataset contains missing (NA) values in columns: {', '.join(missing_cols)}. "
+        
+        if infinite_cols:
+            error_details['infinite'] = infinite_cols
+            error_message += f"Dataset contains infinite (Inf) values in columns: {', '.join(infinite_cols)}. "
+        
+        if error_message:
+            error_message += "These values must be removed or fixed for Robyn to properly work."
+        
+        return ValidationResult(
+            status=not bool(error_details),
+            error_details=error_details,
+            error_message=error_message
+        )
+    
     
     def check_no_variance(self) -> ValidationResult:
         """
         Check for columns with no variance in the input dataframe.
         
         :return: A list of column names with no variance.
         """
-        raise NotImplementedError("Not yet implemented")
+        dt_input = self.mmm_data.data
+        
+        no_variance_cols = dt_input.columns[dt_input.nunique() == 1].tolist()
+        
+        error_details = {}
+        error_message = ""
+        
+        if no_variance_cols:
+            error_details['no_variance'] = no_variance_cols
+            error_message = f"There are {len(no_variance_cols)} column(s) with no-variance: {', '.join(no_variance_cols)}. Please remove the variable(s) to proceed."
+        
+        return ValidationResult(
+            status=not bool(error_details),
+            error_details=error_details,
+            error_message=error_message
+        )
+    
 
     def check_variable_names(self) -> ValidationResult:
         """
         Check variable names for duplicates and invalid characters.
         
         :return: A dictionary with keys 'duplicates' and 'invalid', each containing a list of problematic column names.
         """
-        raise NotImplementedError("Not yet implemented")
+
+        mmmdata_spec = self.mmm_data.mmmdata_spec
+
+        # Collect all variable names to check
+        vars_to_check = [
+            mmmdata_spec.dep_var,
+            mmmdata_spec.date_var,
+            mmmdata_spec.context_vars,
+            mmmdata_spec.paid_media_spends,
+            mmmdata_spec.organic_vars
+        ]
+        vars_to_check = [var for var in vars_to_check if var is not None]
+        vars_to_check = [item for sublist in vars_to_check for item in (sublist if isinstance(sublist, list) else [sublist])]
+
+        # Check for duplicates
+        duplicates = [var for var in set(vars_to_check) if vars_to_check.count(var) > 1]
+
+        # Check for invalid characters (spaces)
+        invalid = [var for var in vars_to_check if ' ' in var]
```

Comment:
What is the definition of invalid here? (do we need regex to cover more invalid cases?)

Reply:
This is the only check R file does for invalid names

Reply:
Let's think beyond current R code...

    invalid_cols = []
    for col in df.columns:
        if not isinstance(col, str):
            invalid_cols.append(col)
        elif not col.isidentifier():
            invalid_cols.append(col)
    return invalid_cols
  
    
    # Example usage
df = pd.DataFrame({
    'valid_col': [1, 2, 3],
    'invalid-col': [4, 5, 6],  # invalid due to hyphen
    '123invalid': [7, 8, 9],  # invalid due to starting with digit
    ('tuple', 'col'): [10, 11, 12]  # invalid due to being a tuple
})

Reply:
Are these rules enforced somewhere? Why cant we have columns with hypen?

Reply:
I am rethinking.. It doesn't matter if it starts with number or has hypen. 1. it should be valid string. 2. " " is also valid string so need to trim it 3. leading and trailing spaces also need be trimmed before checking if it is string.